common/algorithm.h

```diff
@@ -270,5 +270,18 @@ T gcd(T a, T b) {
 #pragma warning(pop)
 #endif
 
+/**
+ * Replacement algorithm for iterables.
+ * Replaces the occurrences of original in [begin, end) with occurrences of replaced 
+ */
+template<class It, class Dat>
+void replace(It begin, It end, const Dat& original, const Dat& replaced) {
```

Comment:
Our formatting convention for references is to have the space before the & sign and not after.
See: http://wiki.scummvm.org/index.php/Code_Formatting_Conventions#Whitespaces